sql/api/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -803,6 +803,26 @@ class Column(val node: ColumnNode) extends Logging with TableValuedFunctionArgum
    */
   def isInCollection(values: java.lang.Iterable[_]): Column = isInCollection(values.asScala)
 
+  /**
+   * A boolean expression that is evaluated to true if the value of this expression is contained
+   * by the provided Dataset/DataFrame.
+   *
+   * @group subquery
+   * @since 4.1.0
+   */
+  def isin(ds: Dataset[_]): Column = {
+    if (ds == null) {
```

Comment:
how does it map to its SQL equivalent?

Reply:
The SQL equivalent is:

```sql
col IN (NULL)
```

For the API, previously `col.isin(null)` invokes `def isin(list: Any*): Column` but now `def isin(ds: Dataset[_]): Column`, so it needs to call the original one explicitly here.

Reply:
Let me add such tests.